.github/workflows/rust.yml

```diff
@@ -22,3 +22,6 @@ jobs:
       run: cargo build --verbose
     - name: Run tests
       run: cargo test --verbose
+    - name: Check clippy
+      if: ${{ always() }}
```

Comment:
Why is this condition necessary?

Reply:
Because it can check the code even if other steps failed for an unknown reason.

Therefore, we don't need to guarantee the tests to check the clippy, or anything like that. If the clippy fails, we can check the fmt and correct both in the same commit

Reply:
Oh that makes sense. Although, it would also run clippy even if `build` failed, right? I think that's ok, but just confirming